prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1409,19 +1409,26 @@ int rxr_ep_init(struct rxr_ep *ep)
 #endif
 
 	if (efa_fork_status == EFA_FORK_SUPPORT_ON)
-		hp_pool_flag = 0;
+		/*
+		 * Use mmap() instead of malloc() for allocating this pool.
+		 * When fork support is on, registering a buffer with ibv_reg_mr will
+		 * set MADV_DONTFORK on the underlying pages.  After fork() the child
+		 * process will have a hole in its heap, and depending on the alignment
+		 * of the allocation, the malloc metadata may be corrupted.
+		 */
+		pkt_pool_flags = OFI_BUFPOOL_MMAP;
 	else
-		hp_pool_flag = OFI_BUFPOOL_HUGEPAGES;
+		pkt_pool_flags = OFI_BUFPOOL_HUGEPAGES;
```

Comment:
We could change this to `OFI_BUFPOOL_HUGEPAGES | OFI_BUFPOOL_MMAP` to keep the allocation in a separate mmap even when fork support is not on, and huge page allocation fails.  Thoughs?

Reply:
Feedback on https://github.com/ofiwg/libfabric/pull/7547 was in favor of fallback to mmap